### PR TITLE
Consul - Add consul tests and upgrade consul to 0.8.x

### PR DESCRIPTION
**Pull request by lachie83**
- Add app version tag
- Update chart maintainers to github username
- Add consul helm tests
- Update documentation
- Remove existing test.sh

### CODE REVIEW

**Comment by lachie83** [comment]
cc @foxish hoping to add more testable charts

File: stable/consul/README.md [hunk 1]
```diff
@@ -48,13 +47,15 @@ The following tables lists the configurable parameters of the consul chart and t
 | `ui.enabled`            | Enable Consul Web UI                  | `false`                                                    |
 | `uiService.enabled`      | Create dedicated Consul Web UI svc    | `false`                                                    |
 | `uiService.type`        | Dedicate Consul Web UI svc type       | `NodePort`                                                 |
+| `test.image`        | Test container image requires kubectl + bash (used for helm test)      | `lachlanevenson/k8s-kubectl`                                                 |
```
Comment (prydonius) [inline stable/consul/README.md]: should this be configurable?

File: stable/consul/templates/NOTES.txt [hunk 2]
```diff
@@ -1,4 +1,6 @@
 1. Watch all cluster members come up.
   $ kubectl get pods --namespace={{ .Release.Namespace }} -w
-2. Confirm consul cluster is healthy
-  $ kubectl exec {{ .Release.Name }}-consul-0 consul members --namespace={{ .Release.Namespace }} | grep server
+2. Test cluster health using Helm test.
+  $ helm test {{ .Release.Namespace }}
```
Comment (prydonius) [inline stable/consul/templates/NOTES.txt]: should be `.Release.Name`

**Review by prydonius — DISMISSED** [review]
Awesome @lachie83, just have some small comments.

File: stable/consul/templates/NOTES.txt (hunk 2 above)
Comment (lachie83) [reply stable/consul/templates/NOTES.txt]: yes. Nice catch!

File: stable/consul/README.md (hunk 1 above)
Comment (lachie83) [reply stable/consul/README.md]: Yes, I think this is fine. I've documented the prerequisites to cut your own container to run the tests if desired

**Comment by lachie83** [comment]
Updated. I think this should be good to go.

**Comment by lachie83** [comment]
@k8s-bot pull-charts-e2e test this

**Comment by lachie83** [comment]
@k8s-bot pull-charts-e2e test this

**Comment by lachie83** [comment]
@prydonius can you please take another pass at this?

**Review by prydonius — APPROVED** [review]
sorry for the delay @lachie83, lgtm!

**Pull request merged**